### PR TITLE
Make behaviour of atomic states consistent wrt type inference

### PR DESCRIPTION
**Pull request by kasper-piskorski**

### CODE REVIEW

**Comment by fppt** [comment]
@kasper-piskorski What's up with this PR? It's quite old and is tagged as WIP?

**Comment by kasper-piskorski** [comment]
I grew more pessimistic about it over time ;p I'm no longer convinced that's the right approach, probably will scrap it

**Comment by fppt** [comment]
Fair enough :)

**Comment by kasper-piskorski** [comment]
ok, no longer WIP but mergeable

File: grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/atom/binary/RelationshipAtom.java [hunk 1]
```diff
@@ -110,7 +109,7 @@
     private ImmutableMultimap<Role, Var> roleVarMap = null;
     private ImmutableMultimap<Role, Type> roleTypeMap = null;
     private ImmutableMultimap<Role, String> roleConceptIdMap = null;
-    private ImmutableList<RelationshipType> possibleRelations = null;
+    private ImmutableList<Type> possibleRelations = null;
```
Comment (aelred) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/reasoner/atom/binary/RelationshipAtom.java]: Is it `Type` because it can include `thing`?

**Pull request merged**